lib/VersionResolver/CompoundSemVerResolver.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Phpactor\VersionResolver;
+
+use Amp\Promise;
+use function Amp\call;
+
+class CompoundSemVerResolver implements SemVersionResolver
```

Comment:
`Aggregate` is a term used most frequently for this type of thing in Phactor I think.